packages/myst-toc/tests/test.yml

```diff
@@ -0,0 +1,69 @@
+title: Table of Contents
+cases:
+  - title: Single file passes
+    content:
+      - file: foo.md 
+  - title: Single file with title passes
+    content:
+      - file: foo.md 
+        title: Foo!
+  - title: Single file with part passes
+    content:
+      - file: foo.md 
+        part: abstract
+  - title: Single file with class passes
+    content:
+      - file: foo.md 
+        class: beta
+  - title: Single file with hidden passes
+    content:
+      - file: foo.md 
+        hidden: true
+  - title: Single file with invalid hidden fails
+    content:
+      - file: foo.md 
+        hidden: "yes"
+    throws: 'The given contents do not form a valid TOC'
```

Comment:
It would be nice to get nicer errors here to show to users. I suspect when we integrate this into myst-frontmatter, we will use the `simple-validators` as that does now, which will allow us to have coercion and better validation messages that work with where we are at now.